core/src/com/unciv/ui/screens/worldscreen/WorldMapHolder.kt

```diff
@@ -206,12 +207,10 @@ class WorldMapHolder(
     }
 
     private fun onTileRightClicked(unit: MapUnit, tile: Tile) {
-        if (UncivGame.Current.gameInfo!!.getCurrentPlayerCivilization().isSpectator()) {
-            return
-        }
         removeUnitActionOverlay()
         selectedTile = tile
         unitMovementPaths.clear()
+        if (!GUI.isAllowedChangeState()) return
```

Comment:
This should definitely use the worldscreen and not the static - we're trying to avoid that as much as possible

Reply:
Reminder :)